### PR TITLE
Implement select farm

### PR DESCRIPTION
**Pull request by MohamedElmdary**

### CODE REVIEW

File: src/components/select_country.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const loading = ref(false)
+const countries = ref<string[]>()
+async function getCountries() {
+  try {
+    loading.value = true
+    const stats = await gridProxyClient.stats.get({ status: NodeStatus.Up })
+    countries.value = Object.keys(stats.nodesDistribution)
+    loading.value = false
+  } catch {
+    setTimeout(getCountries, 1000) /* retry again after 1 sec */
```
Comment (Abohessuin) [inline src/components/select_country.vue:32]: can u tell why making settimeout here

File: src/components/select_farm.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      value.disk === oldValue.disk &&
+      value.publicIp === oldValue.publicIp
+    )
+      return
+
+    shouldBeUpdated.value = true
+  }, 1000)
+)
+
+watch([loading, shouldBeUpdated], async ([l, s]) => {
```
Comment (Abohessuin) [inline src/components/select_farm.vue:99]: can u specify clear names fro variables l,s

File: src/utils/get_farms.ts [hunk 3]
```diff
@@ -1,52 +1,20 @@
...
+export async function getFarms(grid: GridClient, filters: FilterOptions) {
+  const nodes: NodeInfo[][] = []
 
-async function getOnlineFarms() {
-  const nodes = await gridProxyClient.nodes.listAll({ status: NodeStatus.Up })
-  return new Set(nodes.map((node) => node.farmId))
-}
+  let page = 1
+  while (page) {
+    const _nodes = await grid.capacity.filterNodes({ ...filters, page }).catch(() => [])
```
Comment (Abohessuin) [inline src/utils/get_farms.ts:9]: u dont handle catching error here

File: src/utils/get_farms.ts [hunk 4]
```diff
@@ -1,52 +1,20 @@
...
-}
+export async function getFarms(grid: GridClient, filters: FilterOptions) {
+  const nodes: NodeInfo[][] = []
 
-async function getOnlineFarms() {
-  const nodes = await gridProxyClient.nodes.listAll({ status: NodeStatus.Up })
-  return new Set(nodes.map((node) => node.farmId))
-}
+  let page = 1
+  while (page) {
```
Comment (Abohessuin) [inline src/utils/get_farms.ts:8]: why getting all farms why we dont make it like getting like gateway node as this more efficient

**Review by Abohessuin — CHANGES_REQUESTED** [review]
s

File: src/utils/get_farms.ts (hunk 4 above)
Comment (MohamedElmdary) [reply src/utils/get_farms.ts:8]: Currently, We can't filter by farms so it should be updated later when the api is implemented

File: src/utils/get_farms.ts (hunk 3 above)
Comment (MohamedElmdary) [reply src/utils/get_farms.ts:9]: It's already handled

File: src/components/select_farm.vue (hunk 2 above)
Comment (MohamedElmdary) [reply src/components/select_farm.vue:99]: it's named in deps array so it's so obvious

File: src/components/select_country.vue (hunk 1 above)
Comment (MohamedElmdary) [reply src/components/select_country.vue:32]: Delaying it for 1 sec to avoid fast api calls

File: src/utils/get_farms.ts (hunk 4 above)
Comment (Abohessuin) [reply src/utils/get_farms.ts:8]: so will u open issue with it?

File: src/utils/get_farms.ts (hunk 3 above)
Comment (Abohessuin) [reply src/utils/get_farms.ts:9]: yes dont see it

**Pull request merged**